### DISCUSSION TITLE
Help needed on PowerLawCreepStressUpdate class

### DISCUSSION THREAD

**Discussion by ghost** [idaholab/moose#21267] [Q&A Modules: Solid mechanics] [answered]
Dear MOOSE experts,

I’m a newbie to MOOSE, and I’m leveraging the capabilities of the PowerLawCreepStressUpdate class in FALCON (MOOSE application). I have a few questions about the parent classes of the PowerLawCreepStressUpdate.

Line 170 in the SingleVariableReturnMappingSolution.C class reads: scalar = initialGuess (effective_trial_stress).

I think that scalar is being initialized here, but does this mean that scalar is initialized with the “effective_trial_stress ” value? If this is true, does it mean I can define my own parameter that I can use to initialize the scalar parameter?

Also, Line 211 in the RadialReturnStressUpdate.C file reads: _scalar_effective_inelastic_strain = 0.0. Is _scalar_effective_inelastic_strain being initialized here too? If yes, can I define my own value and use it to initialize _scalar_effective_inelastic_strain?

Thanks in advance for your help!

**Comment by jiangwen84** [comment, marked as answer]
No. The `initialGuess` is a a virtual function. By default, it is zero. As I mentioned before in our emails, we use a Newton method to solve `scalar`. A good initial guess will help convergence. If you do not know, zero would be a nice guess.

**Reply by ghost** [reply]
Thanks, @jiangwen84. So what is the role of the effective_trial_stress in the brackets after the initialGuess on **line 170 in the SingleVariableReturnMappingSolution.C** class?

**Reply by jiangwen84** [reply]
It provides a way to formulate the initial guess as a function of effective_trial_stress. In this case, `effective_trial_stress` is not used.
